test/OpenTelemetry.Tests/Metrics/MetricAPITest.cs

```diff
@@ -379,6 +379,77 @@ public void CounterAggregationTest(bool exportDelta)
             }
         }
 
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void CounterAggregationIgnoresNegativeMeasurementsTest(bool exportDelta)
+        {
+            var exportedItems = new List<Metric>();
+
+            using var meter = new Meter($"{Utils.GetCurrentMethodName()}.{exportDelta}");
+            var counterLong = meter.CreateCounter<long>("mycounter");
+            using var meterProvider = Sdk.CreateMeterProviderBuilder()
+                .AddMeter(meter.Name)
+                .AddReader(new BaseExportingMetricReader(new InMemoryExporter<Metric>(exportedItems))
+                {
+                    Temporality = exportDelta ? AggregationTemporality.Delta : AggregationTemporality.Cumulative,
+                })
+                .Build();
+
+            counterLong.Add(10);
+            counterLong.Add(10);
+
+            // -ve numbers are ignored.
+            counterLong.Add(-5);
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+            long sumReceived = GetLongSum(exportedItems);
+            Assert.Equal(20, sumReceived);
+
+            exportedItems.Clear();
+            counterLong.Add(10);
+            counterLong.Add(10);
+
+            // -ve numbers are ignored.
+            counterLong.Add(-15);
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+            sumReceived = GetLongSum(exportedItems);
+            if (exportDelta)
+            {
+                Assert.Equal(20, sumReceived);
+            }
+            else
+            {
+                Assert.Equal(40, sumReceived);
+            }
+
+            exportedItems.Clear();
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+            sumReceived = GetLongSum(exportedItems);
+            if (exportDelta)
+            {
+                Assert.Equal(0, sumReceived);
+            }
+            else
+            {
+                Assert.Equal(40, sumReceived);
+            }
+
+            exportedItems.Clear();
+            counterLong.Add(40);
+            counterLong.Add(20);
+            counterLong.Add(-100);
+            meterProvider.ForceFlush(MaxTimeToAllowForFlush);
+            sumReceived = GetLongSum(exportedItems);
+            if (exportDelta)
+            {
+                Assert.Equal(60, sumReceived);
+            }
+            else
+            {
+                Assert.Equal(100, sumReceived);
+            }
```

Comment:
nit: This check seems superfluous, or is it trying to cover something other than the export being cumulative or delta?

Reply:
exported value is different based on Delta vs Cumulative.

Reply:
ya, the previous round of `.Add` methods covered that case though, so is this needed since I think it just checks the same thing again -- is what I meant to say in my comment